controllers/host/host_controller.go

```diff
@@ -1486,29 +1482,29 @@ func (r *HostReconciler) StopAfterInSync() bool {
 
 // ReconcileExistingHost is responsible for dealing with the provisioning of an
 // existing host.
-func (r *HostReconciler) ReconcileExistingHost(client *gophercloud.ServiceClient, instance *starlingxv1.Host, profile *starlingxv1.HostProfileSpec, host *hosts.Host) error {
+func (r *HostReconciler) ReconcileExistingHost(client *gophercloud.ServiceClient, instance *starlingxv1.Host, profile *starlingxv1.HostProfileSpec, host *hosts.Host, request_namespace string) (error, []error) {
```

Comment:
That's a good point but I thought of having platformnetwork subreconciler's error captured separately so that it will be easier to implement specific exception handling for subreconciler's errors. Unlike other sub-reconcilers, platformnetwork sub-reconciler is not directly involved in host attributes reconciliation so there might be instances where we just want to let certain errors not block host reconciliation. What do you think @yjian118 ?

Reply:
Not quite sure what do you mean "not block host reconciliation", as "inSync = (err == nil && platform_network_subreconciler_errs == nil)" is should be blocking the host status. Would suggest have clear message regarding the reason why it blocks. And you can define the type/struct of an error and guide the error handling by its type